docs/src/components/PageHeader.js

```diff
@@ -31,17 +41,59 @@ export default function ComponentHeader({
   fileName,
   showSourceLink = true,
 }: Props): Node {
+  const { pinnedSection, setPinnedSection } = useNavigationSidebarContext();
+
+  function handlePinnedComponents() {
+    const arrayPinned = JSON.parse(pinnedSection);
+    if (!arrayPinned.includes(fileName || name) && arrayPinned.length < 3) {
+      return setPinnedSection(
```

Comment:
@AlbertCarreras one thing to think about is what happens if a component name changes? E.g. when we changed `GestaltProvider` to `Provider`

Reply:
Mmmm good point

Reply:
We should be able to reset in those cases